python/ray/train/_internal/storage.py

```diff
@@ -449,6 +449,7 @@ def __init__(
         # Invariant: (`storage_filesystem`, `storage_path`) is the location where
         # *all* results can be accessed.
         self.storage_path = storage_path or self.storage_local_path
```

Comment:
Where do the paths get resolved in the `StorageContext`?

Reply:
Don't we only use pyarrow.from_uri() to resolve? I don't think we should resolve the paths in any other way, to avoid issues like this.

Reply:
I think `from_uri` may error for windows paths. @krfricke Are you able to test on windows to see what valid inputs to `from_uri` are?

I think windows path support might need to be a bit messy if pyarrow doesn't handle it well, but hopefully it we can split it off in its own case and not have it affect the rest of the code. Also, windows support is lower priority.

Reply:
I remember the user reporting the issue tested and from_uri() works correctly, but I cannot seem to find that thread..

Reply:
Ah I think this thread: https://github.com/ray-project/ray/pull/36969#issuecomment-1628680226

Looks like we need to normalize it to something like `c:/a/b/c` first.

Reply:
Hmm perhaps the user is just not escaping their backslashes, like `c:\\a\\b\\c` like you need to for Python strings?